From a49e82725ea312b26adc4483ccf92ca6f89e9560 Mon Sep 17 00:00:00 2001
From: Ren Chenglei <chenglei.ren@intel.com>
Date: Fri, 9 Jun 2023 21:54:33 +0530
Subject: [PATCH] INTERNAL: Remove BO_USE_SCANOUT support for I915_TILING_X

On ADL P, BO_USE_SCANOUT is not well supported for I915_TILING_X,
this change is to help remove it on ADL P

Tracked-On: OAM-100001
Signed-off-by: Chenglei Ren <chenglei.ren@intel.com>
---
 i915.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/i915.c b/i915.c
index 246c41f..e03369a 100644
--- a/i915.c
+++ b/i915.c
@@ -226,7 +226,7 @@ static int i915_add_combinations(struct driver *drv)
 {
 	struct i915_device *i915 = drv->priv;
 
-	const uint64_t scanout_and_render = BO_USE_RENDER_MASK | BO_USE_SCANOUT;
+	uint64_t scanout_and_render = BO_USE_RENDER_MASK | BO_USE_SCANOUT;
 	uint64_t render = BO_USE_RENDER_MASK;
 	const uint64_t texture_only = BO_USE_TEXTURE_MASK;
 	// HW protected buffers also need to be scanned out.
@@ -286,6 +286,11 @@ static int i915_add_combinations(struct driver *drv)
 						    .priority = 2,
 						    .modifier = I915_FORMAT_MOD_X_TILED };
 
+#ifdef USE_GRALLOC1
+	if (i915->graphics_version == 12)
+		scanout_and_render = unset_flags(scanout_and_render, BO_USE_SCANOUT);
+#endif
+
 	drv_add_combinations(drv, render_formats, ARRAY_SIZE(render_formats), &metadata_x_tiled,
 			     render_not_linear);
 	drv_add_combinations(drv, scanout_render_formats, ARRAY_SIZE(scanout_render_formats),
-- 
2.40.1

